monitoring/uss_qualifier/scenarios/astm/netrid/common/aggregate_checks.py

```diff
@@ -103,8 +103,35 @@ def run(self):
         self.end_test_step()
         self.end_test_case()
 
+        self.begin_test_case("Verify https is in use")
+        self.begin_test_step("Verify https is in use")
+
+        self._verify_https_everywhere()
+
+        self.end_test_step()
+        self.end_test_case()
+
         self.end_test_scenario()
 
+    def _verify_https_everywhere(self):
+
+        # TODO check if this is the correct way to do things:
+        #  Ie, if we want to define this as a capability and not a failing check if there is no https,
+        #  is it enough to declare a check only if everything went well?
+        #  alternatively, a failure with a severity of LOW might work too?
+
+        found_cleartext_query = False
+        for query in self._queries:
+            if query.request.url.startswith("http://"):
+                found_cleartext_query = True
+                self.record_note("https", f"query is not https: {query.request.url}")
+
+        if not found_cleartext_query:
+            self.check(
+                "All interactions happen over https",
+                self._queries_by_participant.keys(),
+            )
```

Comment:
`self.check` returns a PendingCheck:

```suggestion
        with self.check(
            "All interactions happen over https",
            self._queries_by_participant.keys(),
        ) as check:
            if not found_cleartext_query:
              check.record_failed(...)
```